File: tt/Classes/Player.cs

using tt.Interfaces;
using tt.helpers;

namespace tt.Classes;

public class Player : IPlayer
{
    public string PlayerId { get; set; }
    // Add other properties as needed
    public string User { get; set; }

    public string CurrentGameId { get; set; }

    public bool IsAi  { get; set; }

    public Player(string gameId) 
    {
        PlayerId = GeneratePlayerId.Generate();
        User = GenerateUserName.Generate();
        CurrentGameId = gameId;
        IsAi = false;
    }
}

File: tt/ClientApp/src/app/classes/Player.ts
import { IPlayer} from "../Interfaces/IPlayer";

export class Player implements IPlayer {
    user: string;
    currentGameId: string;
    playerId: string;
    isAi: boolean;
    constructor(user: string, currentGameId: string, playerId: string) {
        this.user = user;
        this.currentGameId = currentGameId;
        this.playerId = playerId;
        this.isAi = false;
        
    }
}